sycl/include/sycl/accessor.hpp

```diff
@@ -3240,6 +3264,35 @@ class __SYCL_EBO host_accessor
     return *this;
   }
 
+  // implicit conversion between const / non-const types for read only accessors
+  template <typename DataT_,
+            typename = std::enable_if_t<
+                IsAccessReadOnly && !std::is_same_v<DataT_, DataT> &&
+                std::is_same_v<std::remove_const_t<DataT_>,
+                               std::remove_const_t<DataT>>>>
+  host_accessor(const host_accessor<DataT_, Dimensions, AccessMode> &other)
+#ifdef __SYCL_DEVICE_ONLY__
+      {}
+#else
+      : host_accessor(other.impl) {
+  }
+#endif // __SYCL_DEVICE_ONLY__
```

Comment:
Nit; I wonder how the formatter would take to
```suggestion
#ifndef __SYCL_DEVICE_ONLY__
      : host_accessor(other.impl)
#endif // __SYCL_DEVICE_ONLY__
       { }
```
Feel free to mark as resolved if the post-formatted code is absolutely ugly.

Reply:
If we're going with this, it should also be applied to the one below.